src/core.js

```diff
@@ -272,6 +272,16 @@ $.extend( $.validator, {
 				this.element( element );
 			}
 		},
+		oninput: function( element ) {
+			if ( element.name in this.submitted || element.name in this.invalid ) {
+				this.element( element );
+			} else {
+				return;
+			}
+		},
+
+		// DEPRECATED
+		// Please use `oninput` instead.
 		onkeyup: function( element, event ) {
```

Comment:
In the usecase you mentioned above, the events that will fire are the `blur` (on first element) and `keyup` (on second element).
Input event is fired synchronously when the value of an element is changed.

> Does the lib work without keyup at all?

Yes and without any problem (as far as I can tell with the manually tests I did myself)